src/Raven.Studio/typescript/viewmodels/resources/resources.ts

```diff
@@ -88,21 +90,109 @@ class resources extends viewModelBase {
 
         // we can't use createNotifications here, as it is called after *resource changes API* is connected, but user
         // can enter this view and never select resource
-        this.addNotification(this.changesContext.globalChangesApi().watchItemsStartingWith("db/", e => this.fetchResources()));
+        this.addNotification(this.changesContext.globalChangesApi().watchItemsStartingWith("db/", (e: globalAlertNotification) => this.fetchResource(e)));
+        //TODO: add notification for fs, cs, ts
 
         return this.fetchResources();
     }
 
+    attached() {
+        super.attached();
+        this.updateHelpLink("Z8DC3Q");
+        ko.postbox.publish("SetRawJSONUrl", appUrl.forDatabasesRawData());
+    }
+
+    // Fetch all resources info
     private fetchResources(): JQueryPromise<resourcesInfo> {
         return new getResourcesCommand()
             .execute()
             .done(info => this.resources(info));
     }
 
-    attached() {
-        super.attached();
-        this.updateHelpLink("Z8DC3Q");
-        ko.postbox.publish("SetRawJSONUrl", appUrl.forDatabasesRawData());
+    // Fetch single resource info
+    private fetchResource(e: globalAlertNotification) {
+
+        // First find if database already exists in the page resources
+        let resource = ko.utils.arrayFirst(this.resources().sortedResources(), (rs: resourceInfo) => {
+            return rs.qualifiedName === e.Id;
+        });
+
+        switch (e.Operation) {
+            case "Write": 
+                if (resource) {
+                    // Relevant for disable/enable...
+                    this.getResourceInfo(resource);
+                } else {
+                    // Add a new resource for a newly created database
+                    let newResourceInfo = this.getNewResource(e.Id.substr(0, 2), e.Id.substr(3));
+                    newResourceInfo.backupEnabled(false);
+                    this.resources().addResource(newResourceInfo); 
+                }
+               
+                break;
+            case "Load":
+                // Database turned ONLINE, get its stats
+                this.getResourceInfo(resource);
+                break;
+            case "Delete":
+                // Delete database from page resources if exists (because maybe another client did the delete..)
+                if (resource) {
+                    this.removeResource(resource);
+                }
+                break;
+        }
+    }
+
+    private getNewResource(resourceType: string, resourceName: string): resourceInfo {
```

Comment:
We have to show it right after write, but let's fetch resource info from server  and then show on UI. 

In general:
1. Listen to both write and loaded events (already done)
2. On event call loadResource
3. After you get response from server update UI
4. If you get -1 as size don't show size or just send from server 0 as size and show 0. In this case we get update with loaded event. Then when we call getResource size won't be zero.